docs/integration/RuoYi.md

```diff
@@ -0,0 +1,201 @@
+```SpringBoot
+@PostMapping("login")
+public R<?> callback(@RequestBody CodeBody code) {//we should define a CodeBody entity which have code and state
+    String token = casdoorAuthService.getOAuthToken(code.getCode(), code.getState());
+    CasdoorUser casdoorUser = casdoorAuthService.parseJwtToken(token);
+    if(casdoorUser.getName()!=null){
+        String casdoorUserName = casdoorUser.getName();
+        if(sysLoginService.getUserByCasdoorName(casdoorUserName)==null){//if database haven't this user
+            // add this user into database
+            sysLoginService.casdoorRegister(casdoorUserName);
+        }
+    }
+    LoginUser userInfo = sysLoginService.casdoorLogin(casdoorUser.getName());//get this user's information by database
+    return R.ok(tokenService.createToken(userInfo));
+}
+```
+In this method,we use casdoor-SpringBoot-sdk method and slightly modified RuoYi-Cloud Method.
+
+For exmaple RuoYi-Cloud original register with account and password,i change the register with account like casdoorRegister.
+
+I also add a method to execute whether this account exists like getUserByCasdoorName and change execute userinfo with account and password to with account.
+
+It's eazy,because we only need delete the part of checking password.
+## Step5. Summary
+### 5.1 font-end 
+- We need delete original login and register.
+- We also need accept code and state and send them to back-end.
+### 5.2 back-end
+RuoYi back-end have perfect login and register function.We just need change a little,so it is very convenient.
+## Step6. Detailed steps
+1. Deploy and configure casdoor.We must take care of organization's password type which should choose bcrypt because
+RuoYi-Cloud's password type is bcrypt.
+2. We should use casdoor syncers to copy database user to your casdoor organization.This step can make the original account
+import to casdoor.
+3. After we deployed casdoor,we should change font-end.We should close RuoYi check code 
+![checkcode Switch](/img/RuoYi-Cloud_loginSwitch.png)
+
+ Note that RuoYi-Cloud checkcode needs close in nacos again.
+ Note that RuoYi-Cloud open register function need change sys.account.registerUser to true. 
+
+4. We should add button jump to casdoor and change data's loginForm
+![login button](/img/RuoYi-Cloud_loginButton.png)
+![data loginForm](/img/RuoYi-Cloud_loginForm.png)
+ Here i write url,you can get url by casdoor-vue-sdk or casdoor-SpringBoot-sdk
+5. Because we don't use original login,we should delete the method about cookie and checkcode method.
+
+ So the new create function:
+```SpringBoot
+created() {
+    let url = window.document.location.href//get url
+    let u = new URL(url);
+    this.loginForm.code = u.searchParams.get('code')//get code and state
+    this.loginForm.state = u.searchParams.get('state')
+    if(this.loginForm.code!=null&&this.loginForm.state!=null){//if code and state is null,execute handleLogin
+      this.handleLogin()
+    }  
+}
```

Comment:
```js
created() {
    let url = window.document.location.href//get url
    let u = new URL(url);
    this.loginForm.code = u.searchParams.get('code')//get code and state
    this.loginForm.state = u.searchParams.get('state')
    if(this.loginForm.code!=null&&this.loginForm.state!=null){//if code and state is null,execute handleLogin
      this.handleLogin()
    }  
}
```

This code also appears [above](https://github.com/casdoor/casdoor-website/pull/294/files#diff-e05b0df1e14be4a7583ad21dc87743ad3bd81f8a5b3f0f132b6272a311b05f74R38), so is it vue code or SpringBoot code?